### PR TITLE
Added redis-rails and setup configuration for it.

### LINKED ISSUE #313: Connect certification to Redis
- [ ] Deploy and configure a redis instance for certification https://github.com/department-of-veterans-affairs/appeals-deployment/issues/136
- [ ] Add redis URL to deployment role (e.g., `REDIS_URL: "{{ redis_url }}"`)
- [ ] Add a redis library to Gemfile
- [ ] Modify [application.rb](https://github.com/department-of-veterans-affairs/caseflow-certification/blob/master/config/application.rb#L30) to use a redis-backed `cache_store`

_Ori edit 9/21 to put checklist in root comment_

### PR DESCRIPTION
**Pull request by nemodjuric**
fixes #313

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ class Application < Rails::Application
     config.autoload_paths += Dir[Rails.root.join('app', 'models', '{**}')]
     config.exceptions_app = self.routes
 
-    config.cache_store = :memory_store, {expires_in: 24.hours, size: 256.megabytes}
+    config.cache_store = :redis_store, Rails.application.secrets.redis_url, { expires_in: 24.hours }
```
Comment (joofsh) [inline config/application.rb]: Looking at the `redis-rails` [README](https://github.com/redis-store/redis-rails#usage), it looks like they namespace the connect url `redis://localhost:6379/0/cache`, that way the cache and other things in the future (like jobs, or session) won't clash. Would it make sense to add that to our setup too?

Comment (nemodjuric) [inline config/application.rb]: @joofsh @orischwartz  I added '0/cache/' to the url.

File: config/application.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ class Application < Rails::Application
     config.autoload_paths += Dir[Rails.root.join('app', 'models', '{**}')]
     config.exceptions_app = self.routes
 
-    config.cache_store = :memory_store, {expires_in: 24.hours, size: 256.megabytes}
+    config.cache_store = :redis_store, Rails.application.secrets.redis_url+'/0/cache/', { expires_in: 24.hours }
```
Comment (askldjd) [inline config/application.rb]: Last week when we created the dev environment, it didn't have the concept of Redis. With this change, `dev` environment will break.

Ideally, `redis_url` should be overridable through environment variables. Also, the prefix of `/0/cache` should also come from env.

@orischwartz suggested to have two env - `REDIS_URL_CACHE` and `REDIS_URL_SIDEKIQ` and we'll configure it properly in deployment.

File: config/secrets.yml [hunk 3]
```diff
@@ -12,13 +12,22 @@
...
 
+  
 staging:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] || "8c6b64e47fa8d523a9ea3d7e7658dbd5ee916003785039b14ffdd076fae644afaedc04eb683fc705c47f6f07020184c8cf3a30e79f2258521a45b861ab8e7862" %>
 # Do not keep production secrets in the repository,
 # instead read values from the environment.
 production:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
+  redis_url_cache: <%= ENV["REDIS_URL_CACHE"] %>
+  redis_url_siqekiq: <%= ENV["REDIS_URL_SIDEKIQ"] %>
```
Comment (joofsh) [inline config/secrets.yml]: typo in key `redis_url_sidekiq`

**Comment by askldjd** [comment]
@nemodjuric Once https://github.com/department-of-veterans-affairs/appeals-deployment/pull/207 is merged, I will approve this. This way, the devop environment will be ready before the Redis change occurs.

**Comment by askldjd** [comment]
dev and uat environment both have REDIS_URL_CACHE and REDIS_URL_SIDEKIQ defined. Once you merge this, let me know how it goes.

**Comment by joofsh** [comment]
@nemodjuric Can we try and get this merged in today please

**Pull request merged**